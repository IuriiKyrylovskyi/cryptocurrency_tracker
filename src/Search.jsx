import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ handleInput }) => {
  return (
    <div className="search">
      <h1 className="text">Search a currency</h1>
      <form>
        <input type="text" placeholder="Search" className="input" onChange={handleInput} />
      </form>
    </div>
  );
};

Search.propTypes = {
  handleInput: PropTypes.func,
};

export default Search;
