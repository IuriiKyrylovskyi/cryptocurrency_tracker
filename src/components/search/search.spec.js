import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search component', () => {
  it('should render Search component', () => {
    const component = shallow(<Search />);
    expect(component).toMatchSnapshot();
  });

  describe('defaultProps', () => {
    it('should use default onChange', () => {
      const result = Search.defaultProps.handleInput();
      expect(result).toBe(undefined);
    });
  });
});
