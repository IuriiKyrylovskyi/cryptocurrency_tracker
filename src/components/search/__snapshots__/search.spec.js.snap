// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render Search component 1`] = `
<div
  className="search"
>
  <h1
    className="text"
  >
    Search a currency
  </h1>
  <form>
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </form>
</div>
`;
